kv_connectors/llmd_fs_backend/src/llmd_fs_backend/worker.py

```diff
@@ -0,0 +1,296 @@
+# limitations under the License.
+
+import math
+import os
+import torch
+from pathlib import Path
+from typing import Optional, Dict
+from vllm.attention.backends.abstract import AttentionBackend
+from vllm.logger import init_logger
+from vllm.v1.kv_offload.worker.worker import OffloadingHandler, TransferSpec, TransferResult
+
+import storage_offload
+
+logger = init_logger(__name__)
+
+# ----------------------------------------------------------------------
+# Base Storage Offloading Handler
+# ----------------------------------------------------------------------
+DEFAULT_MAX_STAGING_MEMORY_GB = 150
+DEFAULT_MAX_THREADS_PER_GPU = 64
+
+class StorageOffloadingHandler(OffloadingHandler):
+    """Base handler with common helpers for Storage offloading."""
+
+    def __init__(
+        self,
+        model_name: str,
+        tp_size: int,
+        tp_rank: int,
+        dtype: torch.dtype,
+        gpu_blocks_per_file: int,
+        threads_per_gpu: int ,
+        attn_backends: dict[str, type[AttentionBackend]] ,
+        max_staging_memory_gb: int = DEFAULT_MAX_STAGING_MEMORY_GB,  # in GB
+        root_dir: str = "/tmp/shared-kv"
+    ):
+
+        self.model_name = model_name
+        self.tp_size = tp_size
+        self.tp_rank = tp_rank
+        self.dtype = dtype
+        self.gpu_blocks_per_file = gpu_blocks_per_file
+        self.base_path = self.get_kv_cache_base_path(model_name, tp_size, tp_rank, dtype, root_dir)
+        self.threads_per_gpu = min(threads_per_gpu , int(os.cpu_count()), DEFAULT_MAX_THREADS_PER_GPU)
+        self.max_staging_memory_gb = max_staging_memory_gb
+        self.h2d_stream = torch.cuda.Stream()
+        self.d2h_stream = torch.cuda.Stream()
+        self.attn_backends = attn_backends
+
+    # ----------------------------
+    # Shared path helpers
+    # ----------------------------
+    @staticmethod
+    def get_kv_cache_base_path(model_name, tp_size, tp_rank, dtype, root_dir: str) -> Path:
+        """Build base path for KV cache storage."""
+        dtype_str = str(dtype).replace("torch.", "")
+        base_path = Path(f"{root_dir}/{model_name}/tp_{tp_size}/rank_{tp_rank}/{dtype_str}")
+        base_path.mkdir(parents=True, exist_ok=True)
+        return base_path
+
+    @staticmethod
+    def get_file_name(base_path: Path, block_hash: int) -> Path:
+        """Return file path for a given block hash."""
+        if isinstance(block_hash, bytes): # convert bytes to int
+            block_hash = int.from_bytes(block_hash, "little")
+        block_hash_hex = f"{block_hash & ((1 << 64) - 1):016x}"
+        subfolder1, subfolder2 = block_hash_hex[:3], block_hash_hex[3:5]
+        full_path = base_path / subfolder1 / subfolder2 / f"{block_hash_hex}.bin"
+        os.makedirs(full_path.parent, exist_ok=True)
+        return full_path
+
+    def compute_buffer_size_mb(self,tensors, gpu_blocks_per_file, layers_before_num_blocks, num_blocks_idx, safety=1.0, min_mb=32, max_mb=None):
```

Comment:
```suggestion
    def compute_buffer_size_mb(self, tensors, gpu_blocks_per_file, layers_before_num_blocks, num_blocks_idx, safety=1.0, min_mb=32, max_mb=None):
```

Please document this function, vLLM style (`Args:` etc.).
Defaults should probably be skipped here.
Also, this reminds me that you have to run vLLM linting on all python files.